explorer/websockethandlers.go

```diff
@@ -96,10 +116,41 @@ func (exp *explorerUI) RootWebsocket(w http.ResponseWriter, r *http.Request) {
 					}
 
 				case "getmempooltxs":
+					exp.pageData.RLock()
 					exp.MempoolData.RLock()
-					msg, err := json.Marshal(exp.MempoolData)
+
+					mempoolVotes := exp.blockData.GetTrimmedMempoolTx(exp.MempoolData.Votes)
+					mempoolTickets := exp.blockData.GetTrimmedMempoolTx(exp.MempoolData.Tickets)
+					mempoolRevs := exp.blockData.GetTrimmedMempoolTx(exp.MempoolData.Revocations)
+					mempoolTxs := exp.blockData.GetTrimmedMempoolTx(exp.MempoolData.Transactions)
```

Comment:
This is all way better than before thanks to the conversion functions you added to eliminate DB access.  However, the code here is identical to the code in `NextHome`.  I suggest factoring out the common code, into something like:

```go
func (exp *explorerUI) MempoolData() *MempoolData {
	// convert the *MempoolInfo in exp.MempoolData to *MempoolData
...
}
```